types/react-bootstrap-table/index.d.ts

```diff
@@ -441,10 +425,10 @@ export interface FooterData {
 	 * The output value from the formatter function will be used instead of the label if the formatter function is
 	 * defined.
 	 */
-	formatter?(tableData: Array<Row<any>>): string | number | ReactElement<any>;
+	formatter?(tableData: any[]): string | number | ReactElement<any>;
 }
 
-export interface SelectRow {
+export interface SelectRow<TRow extends object> {
```

Comment:
Judging from the changes to the tests, since you cannot flow `TRow` from `BootstrapTable` through to `TableHeaderColumn` the generics seem like more of a hindrance than a value. Since there's still some value in the type safety provided by generics, I would recommend adding type parameter defaults for these, such as `export interface SelectRow<TRow extends object = any>`.

Reply:
I've had a look at doing that, but I'm concerned about the result.

Taking the SelectRow as the example, by adding the "= any" you can define it without the type parameter. But if you do define it without the type parameter, the typescript type checker quite merrily accepts completely different items for the TRow parameter for the function parameters in SelectRow, and doesn't tell the user that there may be a problem.

e.g. The following passes typechecking with `export interface SelectRow<TRow extends object = any>`:
```
const selectRow: SelectRow {
  mode: 'checkbox',
  bgColor:
   (row: Job, isSelect: boolean) =>
      (isSelect ? (row.id < 2 ? 'blue' : row.id < 4 ? 'red' : 'yellow') : null),
  onSelectAll:
    (isSelected: boolean, rows: Product[]) =>
      (isSelected ? products.map((row) => row.id) : [])
}
```
Adding the type parameter to SelectRow as either `SelectRow<Product>` or `SelectRow<Job>` results in the type checker correctly identifying that TRow shouldn't be of both types and showing an error. Without the parameter (so that it uses the default "= any") it appears to completely disable the type checking for the generic.

That's why I didn't put the generics on the interfaces that were used to define the fields themselves in the JSX tags, as I found that the "default any" was completely disabling any type safety based on the generic.

Is that really the intent (i.e. to make typechecking opt-in only, without telling the user that they may want to opt-in)? Are there any interfaces in particular where defining the type while creating the object is problematical enough to want to silently disable typechecking for that interface as a default setting?

Reply:
@rbuckton Not sure if I needed to list your name for you to see my questions in response to your review comments.